### PR TITLE
[#8] Initial data modeling [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by n1207n**
This PR addresses to create few initial Django model classes.

### CODE REVIEW

**Comment by lkolacz** [comment]
@n1207n IMO `docker-compose build` is broken on this branch .. instance will not raise .. please check on your env this part .. lunch above comment and tell me about db_1 container .. is it finished on Your env? That bring problems on `django_api` container .. `ImportError: cannot import name User`
instead of above - code looks OK :)

**Comment by n1207n** [comment]
Hi @lkolacz,

Yeah the modeling isn't finished yet. There are some circle dependencies I think. I'm gonna polish off today and look at any additional data models.

**Comment by n1207n** [comment]
@lkolacz, the fix commit is now present.

I recommend you to blow up your db container first then run `docker-compose up`, as InconsistentMigrationHistory error may occur.

**Comment by lkolacz** [comment]
@n1207n Thx! That help, but we had some issues with fronted.. I fix it..

**Comment by lkolacz** [comment]
@n1207n @chachra 
Please check my changes :)
Thx!

File: django_api/django_api/apps/core/models.py [hunk 1]
```diff
@@ -1,23 +1,293 @@
...
+        verbose_name=u'Reference Number',
+        unique=True,
+    )
+    title = models.CharField(max_length=255)
+    status = models.CharField(
+        max_length=3,
+        blank=True,
+        choices=INTERVENTION_STATUS,
+        default=INTERVENTION_STATUS.draft,
+        help_text='Draft = In discussion with partner, '
```
Comment (lkolacz) [inline django_api/django_api/apps/core/models.py]: done

File: django_api/django_api/apps/core/models.py [hunk 2]
```diff
@@ -1,23 +1,293 @@
...
+
+    threshold_tre_usd = models.DecimalField(max_digits=20, decimal_places=4, default=None, null=True)
+    threshold_tae_usd = models.DecimalField(max_digits=20, decimal_places=4, default=None, null=True)
+
+    def __unicode__(self):
+        if self.country_short_code:
+            return "%s (%s)" % (self.name, self.country_short_code)
+        return self.name
+
+    def get_country_name(self):
```
Comment (lkolacz) [inline django_api/django_api/apps/core/models.py]: done

File: django_api/django_api/apps/core/models.py [hunk 3]
```diff
@@ -1,23 +1,293 @@
...
     cluster = models.ForeignKey('cluster.Cluster', related_name="partners")
 
+    class Meta:
+        ordering = ['name']
+        unique_together = ('name', 'vendor_number')
+
+    def __unicode__(self):
+        return self.name
+
+    def get_country(self):
```
Comment (lkolacz) [inline django_api/django_api/apps/core/models.py]: done

**Comment by lkolacz** [comment]
@n1207n I guess we can merge it to develop... please make a last check.. thx
(*btw, this is initial data model so it's obviously that we bring updates during development process)

File: django_api/django_api/apps/reporting/models.py [hunk 4]
```diff
@@ -97,6 +99,8 @@ class Reportable(models.Model):
     blueprint = models.ForeignKey(IndicatorBlueprint, null=True, related_name="reportables")
     objective = models.ForeignKey('cluster.ClusterObjective', null=True, related_name="reportables")
 
+    parent_indicator = TreeForeignKey('self', null=True, blank=True, related_name='children', db_index=True)
```
Comment (n1207n) [inline django_api/django_api/apps/reporting/models.py]: I believe TreeForeignKey is a django-mptt model field reserved for MPTTModel. I'm not sure if we can use this field for normal Django model.

We could use normal ForeignKey('self).

I'll make a change here.

**Pull request closed without merging**